src/row-data-utils.ts

```diff
@@ -193,14 +200,32 @@ class RowDataUtils {
       properties.reqOpts,
     );
     properties.requestData.data = {};
+    // 1. Create a metrics collector.
+    const metricsCollector = new OperationMetricsCollector(
+      properties.requestData.table,
+      MethodName.READ_MODIFY_WRITE_ROW,
+      StreamingState.UNARY,
+      (
+        properties.requestData.table as any
+      ).bigtable._metricsConfigManager!.metricsHandlers,
+    );
+    // 2. Tell the metrics collector an attempt has been started.
+    metricsCollector.onOperationStart();
+    // 3. Make a unary call with gax options that include interceptors. The
+    // interceptors are built from a method that hooks them up to the
+    // metrics collector
     properties.requestData.bigtable.request<google.bigtable.v2.IReadModifyWriteRowResponse>(
       {
         client: 'BigtableClient',
         method: 'readModifyWriteRow',
         reqOpts,
-        gaxOpts: gaxOptions,
+        gaxOpts: withInterceptors(gaxOptions, metricsCollector),
```

Comment:
This feedback probably should have come in the previous PR, but seeing this here, I think it the function would probably be better called `withMetricInterceptor`. The current name makes it seem like you use it to attach any interceptor, but it really is metrics-specific

And by the same logic `interceptor.ts` file should probably called `metric-intercetor.ts`, and location in `src/client-side-metrics` with the other csm-specific code

Reply:
I made these renames